config/webhook/manifests.yaml

```diff
@@ -1,4 +1,33 @@
 
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  creationTimestamp: null
+  name: mutating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  - v1beta1
```

Comment:
Since we [explicitly](https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/master/docs/user-guide/how-to.md#install-or-upgrade-hnc-on-a-cluster) say HNC requires K8s 1.16 or higher, where v1 is introduced, it's okay to remove v1beta1. I'm going to remove all v1beta1 from `admissionReviewVersions`.

Back to your question - I don't know if we still support `v1beta1`. Is it only possible for us to support both `v1` and `v1beta1` if we have a conversion webhook? Does `admissionReviewVersions: v1, v1beta1` mean if v1 is not supported by the apiserver, we need a conversion webhook to convert v1 to v1beta1 for the old apiserver?

From https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/, it says 
> Webhooks are required to support at least one AdmissionReview version understood by the current and previous API server.

Reply:
This is v1beta1 of the admission review objects, not of the HNC API. It could be that controller-runtime actually handles both v1beta1 and v1, and puts whatever it gets into the `admission.Request` object that gets passed to validator. I'm not sure though.

Maybe it's fine to leave this as-is for now and file a new issue to clean this up later.

Reply:
Looks like controller-runtime does accept both v1beta1 and v1, since they're identical: https://github.com/kubernetes-sigs/controller-runtime/pull/1284

Reply:
I still removed v1beta1 for the reason mentioned above:
> Since we explicitly say HNC requires K8s 1.16 or higher, where v1 is introduced, it's okay to remove v1beta1. I'm going to remove all v1beta1 from admissionReviewVersions.

Reply:
sg